addons/assignGear/functions/fnc_assignGearVehicle.sqf

```diff
@@ -76,50 +80,279 @@ if (!isClass _path) exitWith {
+
+                    if (_currentWeight >= _transportBoxWeight) exitWith {
+                        breakTo "add_magazine";
+                    };
+                };
+                // Since, if a bag still exists to add we exist here, we need to re-add it
+                _loadoutInfo set [_chosen, _amounts];
+                _availableMagazines pushBack _chosen;
+            };
+            if (_currentWeight < _transportBoxWeight && _availableItems isNotEqualTo []) then {
+                scopeName "add_item";
+                private _chosen = "";
+                private _amounts = [0];
+                for "_i" from 0 to MAGAZINES_PER_BOX do {
+                    // check if items of this type remain
+                    if ((_amounts select 0) == 0) then {
+                        _loadoutInfo deleteAt _chosen;
+                        if (_availableItems isEqualTo []) exitWith {
+                            breakTo "item_select_loop";
+                        };
+                        _chosen = _availableItems deleteAt 0;
+                        _amounts = _loadoutInfo get _chosen;
+                        _name pushBackUnique ([_amounts select 2] call _getShortName);
+                    };
+                    _amounts params ["_count", "_weight"];
+                    private _toAdd = 1;
+                    if (_weight == 0) then {
+                        _toAdd = _count;
+                    } else {
+                        _toAdd = floor ((_transportBoxWeight - _currentWeight) / _weight);
+                    };
+                    // always add at least one item
+                    _toAdd = 1 max (_toAdd min _count);
+
+                    _currentWeight = _currentWeight + _weight * _toAdd;
+                    _amounts set [0, _count - _toAdd];
+                    _box addItemCargoGlobal [_chosen, _toAdd];
+
+                    if (_currentWeight >= _transportBoxWeight) exitWith {
+                        breakTo "add_item";
+                    };
+                };
+                // Since, if a bag still exists to add we exist here, we need to re-add it
+                _loadoutInfo set [_chosen, _amounts];
+                _availableItems pushBack _chosen;
+            };
+
+            _box setVariable ["ace_cargo_customName", _name joinString ",", true];
+        };
+    };
+    case 3: { // ammo in boxes in vehicle from config
+        private _boxes = "true" configClasses _path;
+        {
+            private _boxType = configName _x;
+            private _boxCount = (getNumber (_x >> "boxCount")) max 1;
+            for "_i" from 1 to _boxCount do {
+                private _box = createVehicle [_boxType, [2,5,105], [], 0, "CAN_COLLIDE"];
+                [_box, _x, ["%1", "%1 " + str _i] select (_boxCount > 1)] call FUNC(setBoxContentsFromConfig);
+                [_box, 1] call ACEFUNC(cargo,setSize);
+                if !([_box, _theVehicle, true] call ACEFUNC(cargo,loadItem)) exitWith {
```

Comment:
I'm not a fan of mission makers needing to set each vehicle cargo size to accommodate this: loadout files should be plug and play across any mission without needing to enter the editor at all. Why not set the box size to zero, or adjust the vehicle cargo space to accommodate the box?

Reply:
Preface this with my personal preference and choice would be just to leave it as is, I think shrinking the size of each container down to 1 is good enough for a mission maker to put boxes in. Especially considering you're probably only putting 3-4 boxes in a truck per coop at most (obviously logi withstanding, but their trucks/vehicle can usually handle more).

> Im not a fan of mission makers needing to set each vehicle cargo size to accommodate this
I can set a hook so they can do similar to #540, where they set the cargo size by config, or would you also like changes there as well? 
> Why not set the box size to zero
I'm not a fan instant put in / take out for boxes and it allows you to take all resupply boxes from a immobile vehicle and pile on in instead of making choices.
> adjust the vehicle cargo space to accommodate the box?
Again I could add a what I mentioned above, but by and large I think vehicle ACE cargo spaces are accurate and reasonable to what should be expected of a vehicle.

Reply:
I agree the insta-unloading is not favourable, but this does introduce barriers to mission makers. A goal of POTATO is to let mission makers have as little resistance as possible when making missions, and this does raise the barrier; imo it raises it enough to be a blocker. When a mission maker decides to add an additional vehicle, they should not have to edit the vehicle properties at all to make it work with the loadouts.

Reply:
Created this PR for ACE, hopefully it's merged. It will allow us to set the size to zero while still allowing some interaction time

https://github.com/acemod/ACE3/pull/10308

Reply:
I'd rather just increase the space, so all a mission maker would need to do is sum the number of boxes. I'll do that through the config like I did in #540 since I don't like opening the door to stacking innumerable boxes into one vehicle.

Reply:
Done here https://github.com/BourbonWarfare/POTATO/pull/544/commits/5da6194249cd40441ba96f2aaa1c9307bd83dc60